### DISCUSSION TITLE
Unable to run tests in VS Code.

### DISCUSSION THREAD

**Discussion by neilghosh** [open-telemetry/opentelemetry-java#3417] [Q&A]
It's not directly related to this project but wanted some help to configure VS Code with it.

I opened the root directory of this project in VS Code and tried to run some of the Unit tests right from the IDE but the Java Test Runner plugin says "No test found" neither the Run/Debug buttons appears near the test methods. Is anyone else using VS Code.

In the Java Projects section I only see the example modules. 

<img width="1790" alt="image" src="https://user-images.githubusercontent.com/726337/126529243-b601aa4d-a3ed-4aeb-9ace-5f47b5856b49.png">

**Comment by jkwatson** [comment]
Does VS code support kotlin build scripts? I think we're all intellij users here.

**Reply by neilghosh** [reply]
I see, even in intellij open I open the project it doesn't show me the structure , neither I am able to mark the test and main directory as source or test source, is there anything obvious I am missing?

![image](https://user-images.githubusercontent.com/726337/126631184-b1d5822e-8030-4638-a213-c20d032e411d.png)

**Reply by jkwatson** [reply]
First question: can you run the gradle build successfully on the command line?

**Reply by neilghosh** [reply]
Yes `./gradlew build` works fine in command line (Mac)

**Reply by jkwatson** [reply]
Ok. How are you opening the project in IntelliJ? What version of IntelliJ are you running?

**Reply by neilghosh** [reply]
I have Ultimate 2021.1.3 and just file -> Open the `opentelemetry-java` java directory. Is there a specific thing you do after opening.

**Reply by neilghosh** [reply]
Ok I managed to see the project heirarchy and even tests were recognized when I downloaded a Fresh CE edition but now I can only run the integration tests but most of the Unit tests are not running with message "No tasks available"

<img width="936" alt="image" src="https://user-images.githubusercontent.com/726337/126749055-c7bfd83a-3b29-48ef-8cc2-9b33878d4892.png">

**Reply by jkwatson** [reply]
I will admit that IntelliJ doesn't always grok the complex structure of the project. I believe we all make sure to run tests via gradle's runner rather than IntelliJ's built in runner. I think this should be the default now, though.

**Reply by neilghosh** [reply]
Got it, So how do I run/debug these tests alone for which it says no tasks available. In intellij setting its already setup runner as gradle.

**Reply by jkwatson** [reply]
I'm not sure what problem you might be having. This "just works" for me in IDEA. What JDK do you have set up in IDEA for the project?

**Reply by jkwatson** [reply]
![image](https://user-images.githubusercontent.com/858731/126805502-88c22517-b3ba-4ce0-9c06-04583133151f.png)

**Reply by jkwatson** [reply]
last resort, though is always "Invalidate caches/restart" and pray!
